tutorials/grid_structure_advanced.ipynb

```diff
@@ -0,0 +1,435 @@
+    "cell_id = np.arange(g.num_cells)\n",
+    "pp.plot_grid(g, cell_value=cell_id, info='cfn', alpha=0.5, figsize=(15,12))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For the rest of this tutorial, we will advice the user to sporadically have a look at this plot. \n",
+    "All the face numbers, node numbers, etc., gathered further down is shown nicely in this figure. \n",
+    "It can therefore be helpful to use as a visual guide. "
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Topological information"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In addition to storing coordinates of cells, faces and nodes, the grid object also keeps track of the relation between them. \n",
+    "Specifically, we can access:\n",
+    "1. The relation between cells and faces\n",
+    "2. The relation between faces and nodes\n",
+    "3. The direction of `face_normals`, as in which of the neighboring cells has the normal vector as outwards pointing.\n",
+    "\n",
+    "Note that there is no notion of edges for 3d grids. \n",
+    "These are not usually needed for the type of numerical methods that are primarily of interest in `porepy`. \n",
+    "The information can still be recovered from the face-node relations, see comments below.\n",
+    "\n",
+    "The topological information is stored in two attributes, `cell_faces` and `face_nodes`. \n",
+    "The latter has the simplest interpretation, so we start out with that one:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 172,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<12x17 sparse matrix of type '<class 'numpy.bool_'>'\n",
+       "\twith 34 stored elements in Compressed Sparse Column format>"
+      ]
+     },
+     "execution_count": 172,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "g.face_nodes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We see that the information is stored as a scipy.sparse matrix. \n",
+    "From the shape of the matrix, we conclude that the rows represent nodes, while the faces are stored in columns. \n",
+    "We can get the nodes for the first face by brute force by writing"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 173,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "array([0, 4], dtype=int64)"
+      ]
+     },
+     "execution_count": 173,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "np.where(g.face_nodes[:, 0].toarray())[0]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "That was hardly elegant, though, and would make for cumbersome implementation of, say, a numerical method. \n",
```

Comment:
I suggest removing this "better approach". It's too technical, even for this tutorial